src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -18,8 +18,14 @@
 
 logger = get_logger(__name__)
 
+# This default cloud name. If you use DummyCli(random_config_dir=True), use get_cloud_config_file() instead.
 CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
 
+
+def get_cloud_config_file(cli_ctx=None):
+    return os.path.join(cli_ctx.config.config_dir, 'clouds.config') if cli_ctx else CLOUD_CONFIG_FILE
```

Comment:
Calling this function multiple times is tedious. Perhaps we simply save it as `cli_ctx.cloud_config`?